sdk/azidentity/chained_token_credential_test.go

```diff
@@ -113,30 +113,103 @@ func TestChainedTokenCredential_GetTokenFail(t *testing.T) {
 	}
 }
 
-type unavailableCredential struct{}
+// TestCredential response
+type testCredentialResponse struct {
+	token *azcore.AccessToken
+	err   error
+}
 
-func (*unavailableCredential) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (token *azcore.AccessToken, err error) {
-	return nil, newCredentialUnavailableError("unavailableCredential", "is unavailable")
+// Credential used for testing
+type TestCredential struct {
+	getTokenCalls int
+	responses     []testCredentialResponse
 }
 
-func TestChainedTokenCredential_GetTokenWithUnavailableCredentialInChain(t *testing.T) {
-	secCred, err := NewClientSecretCredential(fakeTenantID, fakeClientID, secret, nil)
+func (c *TestCredential) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (token *azcore.AccessToken, err error) {
+	index := c.getTokenCalls
+	c.getTokenCalls += 1
+	response := c.responses[index]
+	return response.token, response.err
+}
+
+func testGoodGetTokenResponse(t *testing.T, token *azcore.AccessToken, err error) {
```

Comment:
This is almost the same as `testGetTokenSuccess` in `live_test.go`

Reply:
(We talked over teams and we’ll keep this how it is in this PR at the moment)